pkg/agent/bakerapi.go

```diff
@@ -174,3 +175,11 @@ func findSIGImageConfig(sigConfig datamodel.SIGAzureEnvironmentSpecConfig, distr
 
 	return nil
 }
+
+func (agentBaker *agentBakerImpl) GetCachedVersionsOnVHD() (map[string]datamodel.ProcessedManifest, map[string]datamodel.ContainerImage,
+	map[string]datamodel.DownloadFiles, error) {
+	if datamodel.CachedFromManifest == nil || datamodel.CachedFromComponentContainerImages == nil || datamodel.CachedFromComponentDownloadedFiles == nil {
```

Comment:
should the conditions be && instead of ||?

it looks like right now, if any one of them is nil, we will return nil for all the others as well, that doesn't seem right?

Reply:
I could do this 

```
       cached := datamodel.CachedOnVHD{
		CachedFromManifest:                 datamodel.CachedFromManifest,
		CachedFromComponentContainerImages: datamodel.CachedFromComponentContainerImages,
		CachedFromComponentDownloadedFiles: datamodel.CachedFromComponentDownloadedFiles,
	}

	if datamodel.CachedFromManifest == nil || datamodel.CachedFromComponentContainerImages == nil || datamodel.CachedFromComponentDownloadedFiles == nil {
		return cached, fmt.Errorf("cached versions are not available")
	}

	return cached, nil
```

that way if something exists it will be returned. But I think || makes sense, otherwise it will be a heavy conditional